xemutest/__main__.py

```diff
@@ -12,18 +13,47 @@ def main():
 	ap = argparse.ArgumentParser()
 	ap.add_argument('private', help='Path to private data files')
 	ap.add_argument('results', help='Path to directory where results should go')
+	ap.add_argument('--data', help='Path to test data (e.g., disc images)')
+	ap.add_argument('--xemu', help='Path to the xemu binary')
 	args = ap.parse_args()
 
+	tests = []
 	result = True
-	tests = [Test]
-	for i, test_cls in enumerate(tests):
+	for path in os.listdir(os.path.dirname(__file__)):
+		if not path.startswith("test_") or path == "test_base.py":
+			continue
+		if not path.endswith(".py"):
+			continue
+
+		module = importlib.import_module(path[:-3])
+		for test_name, test_class in inspect.getmembers(module, inspect.isclass):
+			if test_name.startswith("Test"):
+				tests.append((test_name, test_class))
+
+	private_path = os.path.abspath(os.path.expanduser(args.private))
+	results_base = os.path.abspath(os.path.expanduser(args.results))
+	test_data_root = os.path.expanduser(args.data) or os.path.join(os.path.dirname(__file__), 'data')
```

Comment:
This will throw a `TypeError` if `--data` is not set

Reply:
Stupid refactoring mistake; fixed.